modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java

```diff
@@ -141,12 +150,16 @@ public URI getEndpointOverride(Service service) {
             return new URI("http://" +
                 ipAddress +
                 ":" +
-                getMappedPort(service.getPort()));
+                getMappedPort(getServicePort(service)));
         } catch (UnknownHostException | URISyntaxException e) {
             throw new IllegalStateException("Cannot obtain endpoint URL", e);
         }
     }
 
+    private int getServicePort(Service service) {
+        return version.matches(BEFORE_0_11) ? service.port : PORT;
```

Comment:
FYI we have a utility class (`ComparableVersion`) for parsing and comparing version strings. It's probably going to be more reliable than a regex - please could we use that?

Also, there's a slight risk that people may want to use their own custom Localstack image, which could have a completely different form of tag. While I think it's unlikely that this would cause a problem, please could we at least have some logging at time of container startup to give some clues about what's going to happen?

Reply:
Hi, @rnorth 
Thanks for your comment.
* Use `ComparableVersion` - ✅ (although I have to add `isSemanticVersion` method there.
* I've added a warning message if the tag is not "latest" or a semantic version and the code will fall back to the "legacy" mode.
* I've grouped the tests regarding legacy mode into a single class and added some extra test cases.
* I've added an explicit constructor with the "legacy mode" flag - just in case.